src/PersonAuthentication.java

```diff
@@ -0,0 +1,34 @@
+public class PersonAuthentication {
+    public static boolean checkAuthenticationData(String login, String password, String confirmPassword) throws WrongPasswordException, WrongLoginException, WrongPasswordExceptionConfirm {
+        boolean loginLength = login.length() <= 20 && login.length() >= 5;
+        boolean loginMatch = login.matches("[a-zA-Z0-9_]+");
+        boolean passwordLength = password.length() <= 20 && password.length() >= 5;
+        boolean passwordMatch = password.matches("[a-zA-Z0-9_]+");
+        boolean passwordConfirmed = password.equals(confirmPassword);
+
+        if (!loginLength && login.length() < 5) {
```

Comment:
```suggestion
        if (!loginLength) {
```
проверка длины уже учтена выше